### DISCUSSION TITLE
RLS Policy not working (err 42501)

### DISCUSSION THREAD

**Discussion by barrelltech** [supabase/supabase#7251] [General]
Hello - 

I'm trying to implement a basic RLS policy that will only allow a user to interact with records they own.

I've gone through the following steps:

<img width="1198" alt="Screenshot 2022-06-11 at 16 45 44" src="https://user-images.githubusercontent.com/99540338/173192710-748fcba7-7773-4645-a066-6e726e44adfa.png">

<img width="1182" alt="Screenshot 2022-06-11 at 16 45 51" src="https://user-images.githubusercontent.com/99540338/173192713-7bdb8a8a-4765-4921-b412-fa03957bb18f.png">

<img width="1088" alt="Screenshot 2022-06-11 at 16 45 58" src="https://user-images.githubusercontent.com/99540338/173192717-b52dad5c-f90b-4069-a79f-1d5734c15cda.png">

I've confirmed that the data is being set with an `owner-uuid` property

<img width="389" alt="Screenshot 2022-06-11 at 16 49 36" src="https://user-images.githubusercontent.com/99540338/173192769-27019262-3ee2-4f9a-9b54-5a4f8843b83a.png">

I've tried invoking the `insert` method with the options `returning: "minimal"` (although the select call should be valid)

I still get the error `42501`. 

<img width="626" alt="Screenshot 2022-06-11 at 16 50 50" src="https://user-images.githubusercontent.com/99540338/173192823-5c512d4b-f728-4199-a4a7-20e819c77c9f.png">

-- 

I've tried every combination of RLS policies I can think of, explicitly setting the tables, fudging with the data, changing the field name to no include special characters, everything. Nothing seems to be working.

I feel like this should be relatively basic functionality. What am I missing?

PS: Supabase automatically reformats the comparison function after submitting it - I've tried changing it back to auth.uid but it always reverts.

**Comment by GaryAustin1** [comment]
You may not have a user signed in when you make the call.  A quick check would be to set RLS to work for anon and see if insert works.  If so then you are not passing the user token at the time you do the insert call.  Set "Target roles" to anon and policy to true for the test. 

Note: not sure why the UI displays it as uid() instead of auth.uid(), but that is not an issue with the code running correctly.
